src/components/accession2/Accession2GPS.tsx

```diff
@@ -0,0 +1,135 @@
+    right: `-${theme.spacing(4)}`,
+  },
+}));
+
+type Accession2GPSProps = {
+  record: AccessionPostRequestBody;
+  onChange: (id: string, value?: any) => void;
+};
+
+export default function Accession2GPS(props: Accession2GPSProps): JSX.Element {
+  const classes = useStyles();
+  const { record, onChange } = props;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [gpsCoordsList, setGpsCoordsList] = useState<string[]>(['']);
+  const theme = useTheme();
+
+  useEffect(() => {
+    const parsedCoords = gpsCoordsList
+      .filter((coords) => !!coords.trim())
+      .map((coords) => {
+        try {
+          const validCoords = new Coordinates(coords);
+          return {
+            latitude: validCoords.getLatitude(),
+            longitude: validCoords.getLongitude(),
+          } as Geolocation;
+        } catch {
+          // skip invalid coords
+          return null;
+        }
+      })
+      .filter((coords) => coords !== null) as Geolocation[];
+
+    if (!_.isEqual(parsedCoords, record.collectionSiteCoordinates)) {
+      onChange('collectionSiteCoordinates', parsedCoords);
+    }
+  }, [gpsCoordsList, onChange, record.collectionSiteCoordinates]);
+
+  const onUpdateGPS = (value: any, index: number) => {
+    const gps = value as string;
+    setGpsCoordsList((prev) => {
+      const coords = [...prev];
+      coords[index] = gps;
+      return coords;
+    });
+  };
+
+  const onDeleteGpsCoords = (index: number) => {
+    setGpsCoordsList((prev) => {
+      const coords = [...prev];
+      coords.splice(index, 1);
+      return coords;
+    });
+  };
+
+  const onAddGpsCoords = () => {
+    setGpsCoordsList((prev) => {
+      return [...prev, ''];
+    });
+  };
+
+  if (!isOpen) {
+    return (
+      <Grid item xs={12} marginTop={theme.spacing(2)}>
+        <Box display='flex' justifyContent='flex-start'>
+          <Link sx={{ textDecoration: 'none' }} href='#' id='addGPS' onClick={() => setIsOpen(true)}>
+            {strings.ADD_GPS_COORDINATES}
+          </Link>
+        </Box>
+      </Grid>
+    );
+  }
+
+  return (
+    <Box display='flex' flexDirection='column' width='100%' marginTop={theme.spacing(2)}>
+      {gpsCoordsList.map((gpsCoords, index) => (
+        <Box
+          key={index}
+          mb={2}
+          display='flex'
+          alignItems='center'
+          sx={{ display: 'block', position: 'relative', marginBottom: theme.spacing(1) }}
+        >
+          <Textfield
+            id={`gpsCoords${index}`}
+            value={gpsCoords}
+            onChange={(unused, value) => onUpdateGPS(value, index)}
+            label={index === 0 ? strings.LATITUDE_LONGITUDE : ''}
+            type='text'
+          />
+          {index !== 0 && (
+            <IconButton
+              id={`delete-gps-coords${index}`}
+              aria-label='delete'
+              size='small'
+              onClick={() => onDeleteGpsCoords(index)}
+              className={classes.delete}
```

Comment:
Why didn't you use sx prop for this?

Reply:
copy pasta from another component,, will switch to sx